src/dbnode/encoding/m3tsz/timestamp_encoder.go

```diff
@@ -187,6 +187,14 @@ func (enc *TimestampEncoder) writeDeltaOfDeltaTimeUnitUnchanged(
 	}
 
 	deltaOfDelta := xtime.ToNormalizedDuration(curDelta-prevDelta, u)
+	if timeUnit == xtime.Millisecond || timeUnit == xtime.Second {
```

Comment:
It would be useful to have a comment here about why only particular time units are checked